spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -246,11 +270,202 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
+			taskManifest.setSubTaskDeploymentRequests(subTaskAppDeploymentRequests);
+		}
+		else {
+			taskManifest.setTaskDeploymentRequest(appDeploymentRequest);
+		}
+		return taskManifest;
+	}
+
+	/**
+	 * Return the platform specified.  If none have been specified, then "default" will be returned.
+	 *
+	 * @param taskDeploymentProperties properties to interrogate if a platform has been specified
+	 * @return name of the platform configuration to execute the task on
+	 */
+	private String getPlatform(Map<String, String> taskDeploymentProperties) {
+		String platformName = taskDeploymentProperties.get(TASK_PLATFORM_NAME);
+
+		// If not given, use 'default'
+		if (!StringUtils.hasText(platformName)) {
+			platformName = "default";
+		}
+
+		// In case we have exactly one launcher, we override to that
+		List<String> launcherNames = StreamSupport.stream(launcherRepository.findAll().spliterator(), false)
+				.map(Launcher::getName).collect(Collectors.toList());
+
+		if (launcherNames.size() == 1) {
+			platformName = launcherNames.get(0);
+		}
+
+		return platformName;
+	}
+
+	/**
+	 * Determines if the requested deployment is the same (thereby not needing an upgrade) vs different and needing to
+	 * be upgraded via the data in the manifests.  Specifically, the URI of the {@code Resource}, the app properties
+	 * and the deployment properties are all evaluated in this comparison.
+	 *
+	 * @param previousManifest the manifest for the last task execution
+	 * @param newManifest the manifest for the task execution currently being requested
+	 * @return {@code true} if no upgrade is required, {@code false} if an upgrade is required.
+	 */
+	private boolean isAppDeploymentSame(TaskManifest previousManifest, TaskManifest newManifest) {
+		boolean same;
+
+		if(previousManifest == null) {
+			return true;
+		}
+
+		Resource previousResource = previousManifest.getTaskDeploymentRequest().getResource();
+		Resource newResource = newManifest.getTaskDeploymentRequest().getResource();
+
+		try {
+			logger.debug("Previous resource was %s and new resource is %s", previousResource.getURI().toString(), newResource.getURI().toString());
+		}
+		catch (IOException e) {
+			logger.debug("Unable to obtain URIs from resources to be compared in debug log statement", e);
+		}
+
+		same = previousResource.equals(newResource);
+
+		Map<String, String> previousAppProperties = previousManifest.getTaskDeploymentRequest().getDefinition().getProperties();
+		Map<String, String> newAppProperties = newManifest.getTaskDeploymentRequest().getDefinition().getProperties();
+
+		Map<String, String> previousDeploymentProperties = previousManifest.getTaskDeploymentRequest().getDeploymentProperties();
+		Map<String, String> newDeploymentProperties = newManifest.getTaskDeploymentRequest().getDeploymentProperties();
+
+		same = same && previousDeploymentProperties.equals(newDeploymentProperties) && previousAppProperties.equals(newAppProperties);
+
+		return same;
```

Comment:
No because on all platforms, command line args do not require a redeploy.  Local, K8s, and CF all can take a new set of command line args without redeploying the app so we don't need to include them in the check.